### PR TITLE
Mavis consents: record consent from existing cohort parent, instead of creating new parent contact every time

### PR DESCRIPTION
**Pull request by benilovj**
## Pick between cohort parent or new parent

image

## Error: user hits continue without choosing anything

image

## New parent details

image

## Existing parent details

image

## Change links on the confirmation page

image

## Out of scope work

* allow the "who" to be a parent from a previous consent form or consent
* allow the "who" to be the child after they've been assessed as Gillick competent

### CODE REVIEW

File: app/models/consent.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@
 #
 #  id                       :bigint           not null, primary key
 #  health_answers           :jsonb
+#  new_or_existing_parent   :string
```
[the remark this answers is not in the document]
Comment (benilovj) [reply app/models/consent.rb:9]: Yes, no big reasons to keep this field around – I can update the code to use the session to track it...
There will be edge cases where a draft `Consent` is started on one device and continued on another one, but that's not a big consideration I don't think.

[the remark this answers is not in the document]
Comment (tvararu) [reply app/models/consent.rb:9]: Yeah, unless we make draft records very clearly visible in the UI, session seems better in this case

[the remark this answers is not in the document]
Comment (benilovj) [reply app/models/consent.rb]: Ready for re-review - I've replaced saving to the DB with setting a session key

**Review by tvararu — APPROVED** [review]

**Comment by benilovj** [comment]
Oh, looks like I need to fix validations on the new page before merging this.

**Pull request merged**